rfcs/0003-storage.md

```diff
@@ -1,46 +1,107 @@
-2. Any client can use any of these identities to create PoRep proofs. Replicator identities are the CBC encryption keys.
-3. Periodically at a specific PoH count, a replicator that wants to create PoRep proofs signs the PoH hash at that count. That signature is the seed used to pick the block and identity to replicate. A block is 1TB of ledger.
-4. Periodically at a specific PoH count, a replicator submits PoRep proofs for their selected block. A signature of the PoH hash at that count is the seed used to sample the 1TB encrypted block, and hash it. This is done faster than it takes to encrypt the 1TB block with the original identity.
-5. Replicators must submit some number of fake proofs, which they can prove to be fake by providing the seed for the hash result.
-6. Periodically at a specific PoH count, validators sign the hash and use the signature to select the 1TB block that they need to validate. They batch all the identities and proofs and submit approval for all the verified ones.
-7. After #6, replicator client submit the proofs of fake proofs.
+* Validators will randomly sample the set of storage proofs to the set that they can handle, and only the creators of those chosen proofs will be rewarded. The validator can run a benchmark whenever its hardware configuration changes to determine what rate it can validate storage proofs.
+
+# 7. Validation and Replication Protocol
+
+## 7.2 Constants
+
+1. NUM\_STORAGE\_ENTRIES: Number of entries in a block of ledger data. The unit of storage for a replicator.
+2. NUM\_KEY\_ROTATION\_TICKS: Number of ticks to save a PoH value and cause a key generation for the section of ledger just generated and the rotation of another key in the set.
+3. NUM\_STORAGE\_PROOFS: Number of storage proofs required for a storage proof claim to be successfully rewarded.
+4. RATIO\_OF\_FAKE\_PROOFS: Ratio of fake proofs to real proofs that a storage mining proof claim has to contain to be valid for a reward.
+5. NUM\_STORAGE\_SAMPLES: Number of samples required for a storage mining proof.
+6. NUM\_CHACHA\_ROUNDS: Number of encryption rounds performed to generate encrypted state.
+
+## 7.3 Validator behavior
+
+1. Validator joins the network and submits a storage validation capacity transaction which tells the network how many proofs it can process in a given period defined by NUM\_KEY\_ROTATION\_TICKS.
+2. Every NUM\_KEY\_ROTATION\_TICKS the validator stores the PoH value at that height.
+3. Every NUM\_KEY\_ROTATION\_TICKS it also validates samples received from replicators. It signs the PoH hash at that point and uses the low 5 bits of the first byte to index into the signature and then picks a number of bytes of the received sha states with that one in choosing which storage proofs to validate. It then creates a mask of valid proofs and fake proofs and sends it to the leader. This is a storage proof confirmation transaction.
```

Comment:
I don't quite follow this, flagging as a discussion point for next time we sync

Reply:
I added a more detailed set of steps for 7.3.3 in the latest one, hopefully that is clearer.

Reply:
thanks